### ISSUE TITLE
Fusing a single child

### ISSUE THREAD

**Issue by nickeopti** [ComputationalEvolutionaryMorphometry/hyperiax#9]
Something in the `up` execution seems to expect that any node will have at least two children. Consider this example:
```python
import jax
import jax.numpy as jnp

import hyperiax
import hyperiax.execution
import hyperiax.models
import hyperiax.tree

root = hyperiax.tree.TreeNode(data={'state': jnp.zeros(1, dtype=jnp.int32)})
leaf = hyperiax.tree.TreeNode(parent=root, data={'state': jnp.ones(1, dtype=jnp.int32)})
root.children = [leaf]
tree = hyperiax.tree.HypTree(root)

P = jnp.arange(4).reshape(2, 2)

def up(state, **kwargs):
    l = P[:, state]
    jax.debug.print('up: lambda={l}', l=l)
    return {'lambda': l}

def fuse(child_lambda, **kwargs):
    jax.debug.print('fuse: lambda={l}', l=child_lambda)
    return {'alpha': jnp.ones((2, 1))}

up_model = hyperiax.models.UpLambda(up, fuse)
executor = hyperiax.execution.DependencyTreeExecutor(up_model, batch_size=2)
hmm = executor.up(tree)
```
This prints
```
up: lambda=[[[1]]

 [[3]]]
fuse: lambda=[[[1]]]
up: lambda=[[[1]]

 [[3]]]
```
Notice that the `fuse: lambda` step has lost an element. I assume this is due to some broadcasting/indexing which expects multiple elements.

On a side-note, it seems somewhat unnecessary to call the `up` function on the root.

**Comment by frozaken** [comment]
This is intended behaviour, `up` is batched and thus the leading axis on the returned value should be the number of nodes in the batch. I added a more verbose error so it doesn't fail silently in #12

**Comment by frozaken** [comment]
And fwiw you can do it with one note by simply unsqueezing the first axis
